usr/src/uts/common/fs/zfs/arc.c

```diff
@@ -3161,6 +3180,8 @@ arc_evict_state(arc_state_t *state, uint64_t spa, int64_t bytes,
 	}
 	kmem_free(markers, sizeof (*markers) * num_sublists);
```

Comment:
It looks like you didn't indent this (and the 100 lines above) correctly, it should have one more tab.  Or consider moving the code in the new loop to a subroutine.